src/plugins/intel_gpu/src/graph/impls/onednn/utils.cpp

```diff
@@ -150,6 +150,21 @@ void combine_bf_with_first_spatial_dim(cldnn::layout& l) {
     l.set_tensor(t);
 }
 
+void reorder_unreduced_axis_no_fusion(const cldnn::layout& input_layout, cldnn::layout& output_layout, std::vector<uint16_t> axes) {
+    auto in_dims = input_layout.get_tensor().sizes();
+
+    for (size_t idx = 0; idx < axes.size(); idx++) {
+        in_dims[axes[idx]] = 1;
+    }
+
+    auto output_tensor = output_layout.get_tensor();
+    for (size_t idx = 0; idx < in_dims.size(); idx++) {
+        output_tensor.raw[idx] = in_dims[idx];
+    }
+
+    output_layout.set_tensor(output_tensor);
+}
+
```

Comment:
If this is used only in reduce, what about adding it on reduction_onednn.cpp instead?

Reply:
Moved.